smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/PaginationGenerator.java

```diff
@@ -0,0 +1,177 @@
+package software.amazon.smithy.typescript.codegen;
+
+import software.amazon.smithy.codegen.core.CodegenException;
+import software.amazon.smithy.codegen.core.Symbol;
+import software.amazon.smithy.codegen.core.SymbolProvider;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.knowledge.OperationIndex;
+import software.amazon.smithy.model.knowledge.PaginatedIndex;
+import software.amazon.smithy.model.knowledge.PaginationInfo;
+import software.amazon.smithy.model.shapes.OperationShape;
+import software.amazon.smithy.model.shapes.ServiceShape;
+import software.amazon.smithy.model.traits.PaginatedTrait;
+import software.amazon.smithy.typescript.codegen.integration.ProtocolGenerator;
+import software.amazon.smithy.typescript.codegen.integration.RuntimeClientPlugin;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+final class PaginationGenerator implements Runnable {
+
+    private final TypeScriptSettings settings;
+    private final Model model;
+    private final ServiceShape service;
+    private final OperationShape operation;
+    private final SymbolProvider symbolProvider;
+    private final TypeScriptWriter writer;
+    private final Symbol symbol;
+    private final List<RuntimeClientPlugin> runtimePlugins;
+    private final OperationIndex operationIndex;
+    private final Symbol inputType;
+    private final Symbol outputType;
+    private final ProtocolGenerator protocolGenerator;
+    private final ApplicationProtocol applicationProtocol;
+    private final PaginationInfo paginatedInfo;
+
+    private Symbol serviceSymbol;
+    private Symbol operationSymbol;
+    private Symbol inputSymbol;
+    private Symbol outputSymbol;
+
+    private String methodName;
+    private String nonModularServiceName;
+    private String paginationType;
+    private String interfaceLocation;
+
+    PaginationGenerator(TypeScriptSettings settings,
+                        Model model,
+                        OperationShape operation,
+                        SymbolProvider symbolProvider,
+                        TypeScriptWriter writer,
+                        List<RuntimeClientPlugin> runtimePlugins,
```

Comment:
I think theres a bunch in this constructor that we dont actually need. Removing rn